### PR TITLE
Scope non-scoped (global) registries to a single document

### LINKED ISSUE #1390: CustomElementRegistry and adoption
### What is the issue with the DOM Standard?

As part of some work on WebKit @rniwa and I identified a case where we'd like to make a node's custom element registry mutable. We'd like to do this when a node has a CustomElementRegistry whose is global is true and essentially reset it on adoption and cloning into a different document so it gets the global registry of that document, if any, instead.

This would essentially preserve existing behavior and as such I hope it's not controversial. I'm writing some tests and will figure out the specification changes later.

cc @smaug---- @mfreed7 @foolip @AtkinsSJ @keithamus

### PR DESCRIPTION
**Pull request by annevk**
This preserves existing behavior. We throw exceptions in certain corner cases to give us some wiggle room in the future to change some of this around.

Corresponding HTML PR: https://github.com/whatwg/html/pull/11472.

Tests: https://github.com/web-platform-tests/wpt/pull/53791.

Fixes #1390.

- [x] At least two implementers are interested (and none opposed):
   * WebKit
   * Gecko
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/53791
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://issues.chromium.org/issues/399124619 (original Scoped CER bug)
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1874414 (original Scoped CER bug)
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=295337
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: N/A
- [x] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. 

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/dom/1392.html" title="Last updated on Jul 23, 2025, 6:25 AM UTC (c431e0f)">Preview</a> | <a href="https://whatpr.org/dom/1392/1b4bd48...c431e0f.html" title="Last updated on Jul 23, 2025, 6:25 AM UTC (c431e0f)">Diff</a>

### CODE REVIEW

File: dom.bs [hunk 1]
```diff
@@ -7515,8 +7584,17 @@ are:
...
 
- <li><p>If <var>init</var>["{{ShadowRootInit/customElementRegistry}}"] is non-null, then set
- <var>registry</var> to it.
+ <li>
+  <p>If <var>init</var>["{{ShadowRootInit/customElementRegistry}}"] is non-null:
+
+  <ol>
+   <li><p>Set <var>registry</var> to <var>init</var>["{{ShadowRootInit/customElementRegistry}}"].
+
+   <li><p>If <var>registry</var>'s <a for=CustomElementRegistry>is scoped</a> is false and
```
Comment (foolip) [inline dom.bs:7593]: Am I reading this correct that the only thing you can do with `element.attachShadow({customElementRegistry: ...})` is to pass in the document's custom element registry, and anything else will throw an error? That makes it a boolean and practice, but I presume that this is a place where you're reserving some wiggle room for future changes?

**Review by foolip — APPROVED** [review]
Editorially this LGTM. For implementer support I defer to @mfreed7.

File: dom.bs (hunk 1 above)
Comment (rniwa) [reply dom.bs:7593]: you can pass in document's custom element registry or a scoped custom element registry.
